plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/GitHook.kt

```diff
@@ -77,14 +77,20 @@ internal fun generateGitHook(
     echo "Running ktlint over these files:"
     echo "${'$'}CHANGED_FILES"
     
-    git stash push --keep-index
+    git diff > unstaged.diff
```

Comment:
how about moving `unstaged.diff` into `.git/` directory and name it something like `unstaged-ktlint-git-hook.diff` to make it have a little more unique name and reduce possible other hooks name conflict?

Reply:
Probably would be nice additionally introduce some variable to hold diff file name.

Reply:
And better to add `--color=never` diff option, so produced patch could always be consumed by `git apply` command